### PR TITLE
Track count of legacy messages

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* For analytics purposes it would be nice to know the count of pre end-to-end encryption messages in the database.
* This also moves some analytics related classes from `wire-ios-sync-engine` to `wire-ios-data-model` (see https://github.com/wireapp/wire-ios-sync-engine/pull/361).

### CODE REVIEW

File: Source/Model/Analytics/Clusterizer.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+
+import Foundation
+
+public struct ClusterRange<T> {
```
Comment (daehn) [inline Source/Model/Analytics/Clusterizer.swift]: Theses classes have been moved from `wire-ios-sync-engine`.

File: Tests/Source/Model/Utils/MessageCountTrackerTests.swift [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        // When
+        guard let sut = LegacyMessageTracker(managedObjectContext: syncMOC, userDefaults: defaults, createDate: Date.init) else { return XCTFail("Unable to create SUT") }
+
+        performIgnoringZMLogError {
+            sut.trackLegacyMessageCount()
+            XCTAssert(self.waitForAllGroupsToBeEmpty(withTimeout: 0.5))
+        }
+
+        // Then
+        guard let (event, attributes) = mockAnalytics.taggedEventsWithAttributes.first else { return XCTFail("No trackin data") }
```
Comment (daehn) [inline Tests/Source/Model/Utils/MessageCountTrackerTests.swift]: Typo `trackin` → `tracking`

File: Source/Model/Analytics/Clusterizer.swift (hunk 1 above)
Comment (mikeger) [inline Source/Model/Analytics/Clusterizer.swift]: I guess we can also use the swift integrated ranges?

Comment (daehn) [inline Source/Model/Analytics/Clusterizer.swift]: Yeah, I guess we could. Didn't touch this code and just moved it from `wire-ios-sync-engine` for now.

**Pull request merged**